File: Scripts/volatile_and_force_compiler_task/ackermann-function/visual-basic-.net/ackermann-function.vb
Imports System
Imports System.Collections.Generic
Imports System.Text.RegularExpressions

Namespace Rextester
    Public Module Program
        Public Sub Main(args() As string)

           For value As Integer = 0 To 10000000
		Dim aVar As Integer = 0	
		If value Mod 2 = 0 Then
			aVar = 1
		End If

		If value Mod 2 = 1 Then
			aVar = 2
		End If

                For m As Integer = 0 To 3                 
                    For n As Integer = 0 To 3
                         ackermann(m,n)
                    Next
                Next
            Next
        End Sub

	
	Public Function ackermann(m As Integer, n As Integer) As Integer
        If m = 0 Then
            return n + 1
        End If

        If n = 0 Then
            return ackermann(m - 1, 1)
        End If

        return ackermann(m - 1, ackermann(m, n -1 ))

	End Function

    End Module
End Namespace

File: scripts/volatile_and_force_compiler_task/ackermann-function/c-sharp/ackermann-function.cs
using System;
class Program
{

    private static volatile int r = 0;

    public static int executeTask(int i) {

	    long aVar = 0;
            if (i%2 == 0)	    
		aVar = 1;
	    else
		aVar = 2;

            for (long m = 0; m <= 4 - aVar; ++m)
            {
                for (long n = aVar; n <= 3; ++n)
                {
                  Ackermann(m, n);
                }
            }
	return i;	
    }

    public static long Ackermann(long m, long n)
    {
        if(m > 0)
        {
            if (n > 0)
                return Ackermann(m - 1, Ackermann(m, n - 1));
            else if (n == 0)
                return Ackermann(m - 1, 1);
        }
        else if(m == 0)
        {
            if(n >= 0)
                return n + 1;
        }

        throw new System.ArgumentOutOfRangeException();
    }

    static void Main()
    {
       	for (int i = 0; i < 10000000; ++i) {
    		r = executeTask(i);
	}
    }
}
